scripts/home.js

```diff
@@ -0,0 +1,185 @@
+const searchIcon = document.querySelector(".search-icon");
+searchInputField =  document.querySelector(".search-box input[type='text']");
+searchIcon.addEventListener("click",()=>{
+    if(searchBox.classList.contains("hide-search-box")){
+        searchBox.classList.remove("hide-search-box");
+        searchInputField.value = "";
+    }else{
+        searchBox.classList.add("hide-search-box");
+    }
+});  
+
+// prevent page from reload when the search form get submitted 
+document.querySelector(".search-box").addEventListener("submit",event=>{
+    event.preventDefault(); 
+    searchInputField.value = "";
+});
+
+
+// Remove the menu and search box
+// if screen width > 550
+window.onresize = ()=>{
+    if(window.innerWidth > 550){
+        if(menuContent.classList.contains("show")){
+            menuContent.classList.add("hide");
+            menuContent.classList.remove("show");
+        }
+        if(burgerMenu.classList.contains("open")){
+            burgerMenu.classList.remove("open");
+        }
+        searchBox.classList.add("hide-search-box");
+    }
+}  
+
+// This function addes product card to the products grid
+function populateProduct(){
+    productsGrid = document.querySelector("#products .products-grid");
+    products.forEach(product =>{
+        productsGrid.appendChild(createProductCard(product));
+    });  
+} 
+populateProduct(); 
+
+// This function creates product card using its details 
+// It takes a product object as input and 
+// returns a reference to a card div
+function createProductCard(product){
+    const card = document.createElement("div"); 
+    card.classList.add("card");
+
+    const namePrice = document.createElement("div"); 
+    namePrice.classList.add("name-price"); 
+    const name = document.createElement("h3");
+    name.textContent = product.name; 
+    const price = document.createElement("span");
+    price.textContent = product.price+"$";  
+    namePrice.appendChild(name); 
+    namePrice.appendChild(price); 
+
+    const image = document.createElement("img");  
+    image.setAttribute("src",product.imageURL); 
+    image.setAttribute("alt","product image");  
+
+    const addToCart = document.createElement("div");
+    addToCart.classList.add("add-to-cart") 
+    const addToCartImage = document.createElement("img");
+    addToCartImage.setAttribute("src","./images/shopping-cart2.svg");
+    addToCartImage.setAttribute("alt","add to cart image"); 
+    const addToCartText = document.createElement("span");
+    addToCartText.textContent = "Add to cart";   
+    addToCart.appendChild(addToCartImage); 
+    addToCart.appendChild(addToCartText); 
+    
+    card.appendChild(namePrice); 
+    card.appendChild(image); 
+    card.appendChild(addToCart);
+
+
+    return card; 
+}  
+
+// This function generates random number
+// returns a random number
+// uses time object to get unique number to be used 
+// as product id.
+function generateRandomNumber(){
+    return new Date().getTime(); 
```

Comment:
I think if this function is called multiple times in the code (as it is above), there's a good chance that it will return the same value repeatedly (because no time has passed in between the calls).

Reply:
The data above is dummy, in real life scenario, products will be added using (add product form). So, I don't think it will cause a problem.